### PR TITLE
Add C# 9 tests for S3877 , S1067 , S4226 

### PR DESCRIPTION
**Pull request by andrei-epure-sonarsource**
Related to #3668

### CODE REVIEW

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ExtensionMethodShouldBeInSeparateNamespace.CSharp9.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+﻿var x = 1;
```
[the remark this answers is not in the document]
Comment (pavel-mikula-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ExtensionMethodShouldBeInSeparateNamespace.CSharp9.cs:1]: I might see it now, is it a test for namespace in top level statement file?

Can you add this on top level?
```
public static class GlobalExtensions
{
    public static void Bar(this GlobalRecord r) { } // Noncompliant
}
```

[the remark this answers is not in the document]
Comment (pavel-mikula-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ExtensionMethodShouldBeInSeparateNamespace.CSharp9.cs:1]: For easier undertanding:
```suggestion
﻿var typesAndNamespacesInTopLevelFile = true;
```

**Review by pavel-mikula-sonarsource — APPROVED** [review]
LGTM with some minor comments

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ExceptionShouldNotBeThrownFromUnexpectedMethods.CSharp9.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        static Record1() => throw new NotImplementedException();
+        public void Dispose()
+        {
+            throw new NotImplementedException();
+        }
+    }
+
+    record Record2 : IDisposable
+    {
+        static Record2() => throw new Exception(); // FN, needs support for throw expressions
```
[the remark this answers is not in the document]
Comment (andrei-epure-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/ExceptionShouldNotBeThrownFromUnexpectedMethods.CSharp9.cs:17]: I also opened https://github.com/SonarSource/sonar-dotnet/issues/3754

**Pull request merged**